### PR TITLE
Packit rewrite and re-architecture of the miniconda buildpack

### PR DESCRIPTION
**Pull request by ForestEckhardt**
Signed-off-by: Timothy Hitchener <thitchener@pivotal.io>

### CODE REVIEW

File: build.go
Comment (arjun024) [inline build.go:35]: Packit has this released https://github.com/paketo-buildpacks/packit/blob/v0.10.0/postal/service.go#L185

**Review by arjun024 — CHANGES_REQUESTED** [review]
Left a couple of comments. Also would love to see a short rfc about the buildpack design/implementation since this buildpack is a little aytpical (it doesn't download the dependency but downloads an installation script that has to be run), but happy to leave it to your call.
Looks great otherwise

File: build.go
Comment (ForestEckhardt) [reply build.go:91]: We used a layer here because the path for a layer is deterministic and therefore it was easier to make assertions about the path to the artifact that we were downloading the script to. This is also a layer that is guaranteed to be removed by the lifecycle. If this is too objectionable we can remove it.

Comment (arjun024) [reply build.go:91]: That sounds fine. Could you please add that reason as a comment there?

File: constants.go
Comment (arjun024) [inline constants.go:3]: a comment here please

**Comment by arjun024** [comment]
Also, the `.dependabot` dir must be deleted. Dependabot will use the config at `.github/dependabot.yml`.

File: build.go
Comment (arjun024) [reply build.go:76]: @ForestEckhardt The rebuild seems to fail with [this lifecycle error](https://github.com/buildpacks/lifecycle/blob/8d93e588101b22f1e8d3badb9274ebb4bf0894bc/exporter_test.go#L1333-L1349). Can you take a look? If it's not a simple issue to fix, we can file this as a separate issue to investigate (this is a much widespread issue).

#### To reproduce

```
# checkout PR branch

./scripts/package.sh -v 8.8.8

cd integration/testdata/default_app

pack build minimage -b ../../../build/buildpackage.cnb -b gcr.io/paketo-community/build-plan
# Build succeeds

# now add build=true to plan.toml
cat plan.toml
[[requires]]
  name = "conda"

  [requires.metadata]
    launch = true
    build = true

pack build minimage -b ../../../build/buildpackage.cnb -b gcr.io/paketo-community/build-plan
===> DETECTING
[detector] paketo-community/miniconda  8.8.8
[detector] paketo-community/build-plan 0.0.11
===> ANALYZING
[analyzer] Restoring metadata for "paketo-community/miniconda:conda" from app image
===> RESTORING
===> BUILDING
[builder] Paketo Miniconda Buildpack 8.8.8
[builder]   Reusing cached layer /layers/paketo-community_miniconda/conda
[builder]
===> EXPORTING
[exporter] ERROR: failed to export: layer 'paketo-community/miniconda:conda' is cache=true but has no contents
```

Comment (arjun024) [reply build.go:76]: Given the non-trivial nature of this issue, I'm gonna create a separate issue for this and merge this PR once integration tests pass

**Comment by arjun024** [comment]
I tried re-triggering but the integration tests go on forever.

**Pull request closed without merging**
